lib/ui/window.dart

```diff
@@ -294,23 +357,24 @@ class Locale {
       return false;
     final Locale typedOther = other;
     return languageCode == typedOther.languageCode
-        && countryCode == typedOther.countryCode;
+        && scriptCode == typedOther.scriptCode
+        && countryCode == typedOther.countryCode
+        && _variants == typedOther._variants;
   }
 
   @override
-  int get hashCode {
-    int result = 373;
-    result = 37 * result + languageCode.hashCode;
-    if (_countryCode != null)
-      result = 37 * result + countryCode.hashCode;
-    return result;
-  }
+  int get hashCode => hashValues(languageCode, scriptCode, countryCode, hashList(_variants));
 
   @override
   String toString() {
```

Comment:
Under discussion, taking it as out of scope for this Pull Request though, to not block: toString() being for "debug" output, when StringBuffer() makes it functionally useful.

Reply:
StringBuffer is basically the _only_ class (of hundreds if not thousands) that makes it useful for non-debugging purposes.